Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -229,4 +230,33 @@ Value ArrayPrototype::join(Interpreter& interpreter)
     return join_array_with_separator(interpreter, *array, separator);
 }
 
+Value ArrayPrototype::concat(Interpreter& interpreter)
+{
+    auto* array = array_from(interpreter);
+    if (!array)
+        return {};
+
+    auto* new_array = interpreter.heap().allocate<Array>();
+    auto& start_array = *array->elements().begin();
+    auto* start_array_values = const_cast<const Value*>(&start_array);
+    new_array->elements().append(start_array_values, array->elements().size());
+
+    for (size_t i = 0; i < interpreter.argument_count(); ++i) {
+        auto argument = interpreter.argument(i);
+        if (argument.is_array()) {
+            auto* argument_object = argument.to_object(interpreter.heap());
+            auto* argument_array = static_cast<Array*>(argument_object);
+
+            auto& arg_start_array = *argument_array->elements().begin();
+            auto* arg_start_array_values = const_cast<const Value*>(&arg_start_array);
+
+            new_array->elements().append(arg_start_array_values, argument_array->elements().size());
```

Comment:
Can we just do this with `Vector::append(const Vector&)` instead?:
```suggestion
            new_array->elements().append(argument_array.elements());
```

Reply:
Alright then, I just did it because I thought about reducing the grow_capacity calls

Reply:
`Vector::append(Vector)` only does a single grow_capacity :)

Reply:
Yes, I got confused :)